src/jvmMain/kotlin/com/charleskorn/kaml/YamlOutput.kt

```diff
@@ -90,6 +92,23 @@ internal class YamlOutput(
     private fun emitQuotedScalar(value: String, scalarStyle: ScalarStyle) = emitScalar(value, scalarStyle)
 
     override fun encodeElement(descriptor: SerialDescriptor, index: Int): Boolean {
+        if(configuration.serializeComments) {
+            var comment = descriptor.getElementAnnotations(index)
+                .filterIsInstance<YamlComment>()
+                .firstOrNull()
+                ?.comment
+
+            if(comment != null) {
+                // Is multiline comment?
+                if(comment.indexOfAny(charArrayOf('\n', '\r')) > -1)
+                    comment = comment.trimIndent()
```

Comment:
In the interests of consistency, should we just always `trimIndent` the comment? Then any leading spaces are always removed, regardless of whether the comment is one or multiple lines.